multiqc/modules/bcl2fastq/bcl2fastq.py

```diff
@@ -0,0 +1,138 @@
+        super(MultiqcModule, self).__init__(name='bcl2fastq', anchor='bcl2fastq',
+        href="https://support.illumina.com/downloads/bcl2fastq-conversion-software-v2-18.html",
+        info="can be used to both demultiplex data and convert BCL files to FASTQ file formats for downstream analysis.")
+
+        # Gather data from all json files
+        self.bcl2fastq_data = dict()
+        for myfile in self.find_log_files('bcl2fastq'):
+            self.parse_file_as_json(myfile)
+
+        # Collect counts by lane and sample (+source_files)
+        self.bcl2fastq_bylane = dict()
+        self.bcl2fastq_bysample = dict()
+        self.source_files = dict()
+        self.split_data_by_lane_and_sample()
+
+        # Filter to strip out ignored sample names
+        self.bcl2fastq_bylane = self.ignore_samples(self.bcl2fastq_bylane)
+        self.bcl2fastq_bysample = self.ignore_samples(self.bcl2fastq_bysample)
+
+        # Return with Warning if no files are found
+        if len(self.bcl2fastq_bylane) == 0 and len(self.bcl2fastq_bysample) == 0:
+            log.debug("Could not find any bcl2fastq data in {}".format(config.analysis_dir))
+            raise UserWarning
+
+        # Print source files
+        for s in self.source_files.keys():
+            self.add_data_source(s_name=s, source=",".join(list(set(self.source_files[s]))), module='bcl2fastq', section='bcl2fastq-bysample')
+
+        # Add sample counts to general stats table
+        self.add_general_stats()
+        self.write_data_file({str(k): self.bcl2fastq_bylane[k] for k in self.bcl2fastq_bylane.keys()}, 'multiqc_bcl2fastq_bylane')
+        self.write_data_file(self.bcl2fastq_bysample, 'multiqc_bcl2fastq_bysample')
+
+        # Add section for counts by lane
+        self.add_section (
+            name = 'bcl2fastq by lane',
+            anchor = 'bcl2fastq-bylane',
+            description = 'Number of reads per lane (with number of perfect index reads)',
+            helptext = "Perfect index reads are those that do not have a single mismatch. All samples of a lane are combinned. Undetermined reads are treated as a third category. To avoid conflicts the runId is prepended.",
+            plot = bargraph.plot(self.get_bar_data_from_counts(self.bcl2fastq_bylane))
+        )
+
+        # Add section for counts by sample
+        self.add_section (
+            name = 'bcl2fastq by sample',
+            anchor = 'bcl2fastq-bysample',
+            description = 'Number of reads per sample (with number of perfect index reads)',
+            helptext = "Perfect index reads are those that do not have a single mismatch. All samples are aggregated across lanes combinned. Undetermined reads are ignored. Undetermined reads are treated as a separate sample. To avoid conflicts the runId is prepended.",
+            plot = bargraph.plot(self.get_bar_data_from_counts(self.bcl2fastq_bysample))
+        )
+
+    def parse_file_as_json(self, myfile):
+        content = json.loads(myfile["f"])
```

Comment:
Yeah, would be good to have a `log.warn` with the filename saying that it was invalid JSON.